### DISCUSSION TITLE
What is the simple way to control the string generated by ContentType?

### DISCUSSION THREAD

**Discussion by Magicloud** [rwf2/Rocket#2875] [Questions]
I am writing a webhook tool for a Go application, which checks the response content type via direct string equality.

And when I used `ContentType::new("application", "json").with_params(("version", "1"))` to set it, the result was "application/external.dns.webhook+json; version=1", which failed the comparison since in the Go code, there is no space before "version".

I wonder if there is a simple way I can remove the space.

**Comment by raif-s-naffah** [comment]
hi @Magicloud ,

have you tried setting it directly in the Response?  there's an [example in the documentation][1] on how you can do that.

[1]: https://api.rocket.rs/v0.5/rocket/struct.Response#example-9

**Reply by Magicloud** [reply]
Yes, it worked. Just that we have the ContentType and everything, if we can do this within its system.

**Reply by the10thWiz** [reply]
I believe may actually be a bug in your Go tool. Based on the MIME type RFC (https://datatracker.ietf.org/doc/html/rfc2045#section-5.1), whitespace is allowed between parameters, but is not considered part of the parameter identifier. Therefore, according to the spec, `text/plain; charset=UTF-8`and `text/plain;charset=UTF-8` should be considered identical content-types.

**Reply by Magicloud** [reply]
Ye, I checked the Go code, it is a string comparison. Anything technically allowed but not the same as the string would fail....
